### PR TITLE
Add multiple location support for creation page and delete preset feature.

### PR DESCRIPTION
**Pull request by neelbagora**
For this PR, I added code supporting multiple targeting locations (only US states) and multiple negative locations (not functional yet). I also added support for multiple locations in presets. Additionally, I added a delete preset function and a 20 preset limit. Aside from that, just small changes to the page (Ex: added contact page link in side menu and added random generation for impressions, clicks, costs).

### CODE REVIEW

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js [hunk 1]
```diff
@@ -292,9 +292,15 @@ function sendFormData() {
...
   keyvalPairs.push(encodeURIComponent('DSACampaignId') + '=' + encodeURIComponent('0'));
   keyvalPairs.push(encodeURIComponent('campaignStatus') + '=' + encodeURIComponent('pending'));
-  keyvalPairs.push(encodeURIComponent('clicks') + '=' + encodeURIComponent('0'));
-  keyvalPairs.push(encodeURIComponent('cost') + '=' + encodeURIComponent('0'));
-  keyvalPairs.push(encodeURIComponent('impressions') + '=' + encodeURIComponent('0'));
+
+  /* temporary random generation for data */
+  let clicks = Math.ceil(Math.random() * 500000);
+  let cost = Math.random() * 23000;
+  let impressions = Math.ceil(Math.random() * 40000);
```
Comment (nevingeorge) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js:338]: Can you do this instead -
let impressions = Math.round(1000 + Math.random()*1000);
let clicks = Math.round(250 + Math.random()*250);
let cost = Math.round(100 + Math.random()*100);
I think these are more accurate daily estimates (impressions should be > clicks). Also, for display purposes the numbers should be closer to each other.

File: DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/DeletePresetServlet.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC
```
Comment (nevingeorge) [inline DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/DeletePresetServlet.java:1]: Can you add a test for the doPost function?

Comment (neelbagora) [reply DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/DeletePresetServlet.java:1]: Yup, will add that after our sync.

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js (hunk 1 above)
Comment (neelbagora) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js:338]: Sounds good.

**Comment by nevingeorge** [comment]
This PR shouldn't change DSACampaignsServlet.java, the merge conflict might be because of the PR I just pushed.

File: DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/DeletePresetServlet.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.lang.*;
+import java.util.ArrayList;
+
+// This servlet is responsible for deleting campaign preset data.
+@WebServlet("/delete-preset")
+public class DeletePresetServlet extends HttpServlet {
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/DeletePresetServlet.java:37]: Why not just merge this into the existing preset servlet?

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js [hunk 4]
```diff
@@ -20,8 +20,20 @@ let keywordCampaignId = null;
...
+
 // array to keep track of all preset names belonging to user
-const userPresets = [];
+const USERPRESETS = [];
+
+// length of mm-dd-yyyy (error handling)
+const DATE_LENGTH = 10;
+
+// number of max presets (to prevent random page behavior)
+const MAX_PRESETS = 20;
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js:36]: Why would there be random page behavior?

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js [hunk 5]
```diff
@@ -20,8 +20,20 @@ let keywordCampaignId = null;
...
 
+// number of negative locations
+let negLocationCount = 1;
+
+// preset index selected.
+let selectedPreset = -1;
+
 // array to keep track of all preset names belonging to user
-const userPresets = [];
+const USERPRESETS = [];
```
Comment (avenkat-google) [inline DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js:30]: nit: USER_PRESETS

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js (hunk 4 above)
Comment (neelbagora) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js:36]: When there are too many presets, it extends the page beyond the form and makes it unorganized

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js (hunk 5 above)
Comment (neelbagora) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js]: Fixed

File: DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/DeletePresetServlet.java (hunk 3 above)
Comment (neelbagora) [reply DSA-Campaign-Uplift-Estimation/src/main/java/com/google/sps/servlets/DeletePresetServlet.java:37]: I'll change it and push it in a bit

File: DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js (hunk 4 above)
Comment (neelbagora) [reply DSA-Campaign-Uplift-Estimation/src/main/webapp/Create/create-script.js:36]: Changed comment to be more clear.

**Comment by neelbagora** [comment]
Some of the linter errors don't really make sense so I avoided them for now.

**Pull request merged**